src/storagetm/storagetm.c

```diff
@@ -950,9 +950,11 @@ static int device_added(Context *c, sd_device *device) {
                 .st_mode = S_IFBLK,
         };
 
-        r = sd_device_get_devnum(device, &lookup_key.st_rdev);
+        dev_t devnum;
+        r = sd_device_get_devnum(device, &devnum);
         if (r < 0)
                 return log_device_error_errno(device, r, "Failed to get major/minor from device: %m");
+        lookup_key.st_rdev = devnum;
```

Comment:
man, this mips abi is total garbage, can't the porters who care about this arch get this fixed properly? it's such a mess that this is carried on year over year and has to be patched in myriad of codebases and not a the actual place that caused this mess...

Anyway, please add an explicit comment here somewhere that explains why this indirection is done here, otherwise this will break again once somebody later sees this and decides it's an unnecessary complication that can be dropped, without the context.

Also, please link the relevant MIPS glibc bug report, so that people know where to check to see if things changed and this hack can be dropped again.

Reply:
Added a comment at both places.